### ISSUE TITLE
Some (all?) repositories are marked dirty on windows

### ISSUE THREAD

**Issue by ThibautVerron** [raxod502/straight.el#469]
Hi,

I have been trying to run `straight-freeze-versions`, and I have run into some problems.

One of them is on a windows machine: I had a lot (maybe all) repositories marked dirty, with all their files modified. I did not modify them, so I discarded the changes.

I suspect it might be due to line endings differences, which git tries to harmonize (!). 
I have git set globally to do nothing with the line endings (emacs can edit unix line endings alright), it might contribute to the problem.

Stupidly I did not think of checking a diff before discarding all the changes, so I do not know for sure that line endings are the problem. But it sounds like the only reasonable explanation, doesn't it?

**Comment by ThibautVerron** [comment]
For some reason I did not get the bug/feature/support message when submitting this issue, sorry.

**Comment by raxod502** [comment]
`straight.el` does not pass any special options to Git with regard to line endings so I'm afraid this seems like a system configuration problem to me. It is of course a possibility to make `straight.el` override Git's configuration settings to avoid this issue, though, since Git's documentation on the line-ending options is rather impenetrable.

I don't think I can do anything more for you without a way to reproduce the issue.

**Comment by ThibautVerron** [comment]
Actually, on second thought I think I changed the git setting _after_ cloning the packages. It's predictable that it would lead to problems, just like I had to confirm similar changes on work repositories.

But that could still be an annoying problem for users with hundreds of packages. Maybe a catch-all solution for this kind of problems would be an "apply this choice for the remaining problems" option in the prompt? If there was already something like that, sorry I missed it.
